server/app/services/geo/esri/RealEsriClient.java

```diff
@@ -210,24 +214,21 @@ && hasCandidatesArray(rootNode)) {
                 // Reason for doing this is to not make more external calls than are needed. This
                 // is both for performance and billing reasons.
                 boolean hasAnyNodesUnderTheScoreThreshold =
-                    StreamSupport.stream(rootNode.get(CANDIDATES_NODE_NAME).spliterator(), false)
```

Comment:
Behold, the spliterator. The source cause of the edge case bug that triggered the cleanup.